website/addons/s3/static/tests/s3NodeConfig.test.js

```diff
@@ -0,0 +1,186 @@
+/*global describe, it, expect, example, before, after, beforeEach, afterEach, mocha, sinon*/
+'use strict';
+var $ = require('jquery');
+
+var assert = require('chai').assert;
+
+var utils = require('tests/utils');
+var faker = require('faker');
+
+var s3NodeConfigVM = require('../s3NodeConfig')._S3NodeConfigViewModel;
+
+var API_BASE = '/api/v1/12345/s3';
+var makeSettingsEndpoint = function(result, urls) {
+    var settingsUrl = [API_BASE, 'settings', ''].join('/');
+    return {
+        method: 'GET',
+        url: settingsUrl,
+        response: {
+            result: $.extend({}, {
+                current_bucket: faker.internet.domainWord(),
+                urls: $.extend({}, {
+                    create_bucket: [API_BASE, 'newbucket', ''].join('/'),
+                    import_auth: [API_BASE, 'import-auth', ''].join('/'),
+                    create_auth: [API_BASE, 'authorize', ''].join('/'),
+                    deauthorize: settingsUrl,
+                    bucket_list: [API_BASE, 'buckets', ''].join('/'),
+                    set_bucket: settingsUrl,
+                    settings: '/settings/addons/'
+                }, urls)
+            }, result)
+        }
+    };
+};
+
+var noop = function() {};
+var Case = function(cfg) {
+    this.description = cfg.description || '';
+    this.endpoint = cfg.endpoint || {
+        response: {
+            result: {}
+        }
+    };
+    this.expected = this.endpoint.response.result;
+    $.extend(this.expected, cfg.data || {});
+};
+Case.prototype.run = function(test) {
+    var server;
+    this.before = function() {
+        server = utils.createServer(sinon, [this.endpoint]);
+    }.bind(this);
+    this.after = function() {
+        server.restore();
+    }.bind(this);
+    test(this);
+};
+var Cases = function(test, cases) {
+    for (var i = 0; i < cases.length; i++) {
+        new Case(cases[i]).run(test);
+    }
+};
+
+describe('s3NodeConfigViewModel', () => {
+    new Cases(
```

Comment:
@sloria, when you have time can you review this pattern for generating test cases?